src/modules/flow/iio/nodes.c

```diff
@@ -776,12 +776,16 @@ humidity_open(struct sol_flow_node *node, void *data, const struct sol_flow_node
 }
 
 static bool
-adc_create_channels(struct iio_double_data *mdata, int device_id)
+adc_create_channels(struct iio_double_data *mdata, int device_id, int channel_id)
 {
+    char channel_name[NAME_MAX];
+
+    snprintf(channel_name, sizeof(channel_name), "in_voltage%d", channel_id);
```

Comment:
Maybe it's different for ADC devices, but couldn't we have a channel named only `in_voltage`, with no index? This is so for proximity sensors, for instance. So, maybe '-1' should mean 'no index' instead of 'first index'. If common case is to have index, default could be '0', then.